packages/scrollbar/__tests__/scrollbar.spec.ts

```diff
@@ -16,25 +11,17 @@ const _mount = (template: string) => mount({
 
 describe('ScrollBar', () => {
   test('vertical', async () => {
-    const outterHeight = 200
+    const outerHeight = 200
     const innerHeight = 500
     const wrapper = _mount(`
-      <el-scrollbar style="height: ${outterHeight}px">
+      <el-scrollbar style="height: ${outerHeight}px">
         <div style="height: ${innerHeight}px;"></div>
       </el-scrollbar>
-`)
-    Object.defineProperties(window.HTMLElement.prototype, {
-      clientHeight: {
-        get () {
-          return outterHeight
-        },
-      },
-      scrollHeight:{
-        get () {
-          return innerHeight
-        },
-      },
-    })
+    `)
+
+    defineGetter(window.HTMLElement.prototype, 'clientHeight', outerHeight)
+    defineGetter(window.HTMLElement.prototype, 'scrollHeight', innerHeight)
```

Comment:
You need to restore these after this test gets finished otherwise it might compromise other tests

Reply:
`clientHeight` can use `jest.spyOn` to mock a value.

Reply:
When I test `Slider` component, I use `jest.spyOn` to mock the `clientHeight` and the return of `getBoundingClientRect()`. I extra it as a test util function in `utils/jestMock.ts`

The code just like:
```
// mock value
const mockClientWidth = jestSpyOn(
      wrapper.find('.el-slider__runway').element,
      'clientWidth',
      () => 200,
      'get')
    const slider:any = wrapper.findComponent({ name:'ElSliderButton' })
    nextTick().then(() => {
      slider.trigger('mousedown',{ clientX: 0 })
      const mousemove = document.createEvent('MouseEvent')
      mousemove.initMouseEvent('mousemove', true, true, window, 1, 100, 0, 100, 0, false, false, true, false, 0, null)
      window.dispatchEvent(mousemove)
      const mouseup = document.createEvent('MouseEvent')
      mouseup.initMouseEvent('mouseup', true, true, window, 1, 100, 0, 100, 0, false, false, true, false, 0, null)
      window.dispatchEvent(mouseup)
      setTimeout(() => {
        expect(wrapper.vm.value === 0.5).toBeTruthy()
        // restore value
        mockClientWidth.mockRestore()
        done()
      }, 10)
    })
```

Reply:
> When I test `Slider` component, I use `jest.spyOn` to mock the `clientHeight` and the return of `getBoundingClientRect()`. I extra it as a test util function in `utils/jestMock.ts`
> 
> The code just like:
> 
> ```
>     // mock value
>     const mockClientWidth = jestSpyOn(
>       wrapper.find('.el-slider__runway').element,
>       'clientWidth',
>       () => 200,
>       'get')
> 
>     const slider:any = wrapper.findComponent({ name:'ElSliderButton' })
>     nextTick().then(() => {
>       slider.trigger('mousedown',{ clientX: 0 })
>       const mousemove = document.createEvent('MouseEvent')
>       mousemove.initMouseEvent('mousemove', true, true, window, 1, 100, 0, 100, 0, false, false, true, false, 0, null)
>       window.dispatchEvent(mousemove)
>       const mouseup = document.createEvent('MouseEvent')
>       mouseup.initMouseEvent('mouseup', true, true, window, 1, 100, 0, 100, 0, false, false, true, false, 0, null)
>       window.dispatchEvent(mouseup)
>       setTimeout(() => {
>         expect(wrapper.vm.value === 0.5).toBeTruthy()
>         
>         // restore value
>         mockClientWidth.mockRestore()
>         done()
>       }, 10)
>     })
> ```

It's a good idea to use `jest.spyOn`, but I don't think it is necessary to extract `jestSpyOn`.